### DISCUSSION TITLE
Limit number of sumultanious connections per device

### DISCUSSION THREAD

**Discussion by JustKeonix** [Ylianst/MeshCentral#5844] [Q&A]
Is there a way to restrict a device on MeshCentral to only one user connection at a time, with exceptions for admins? We're setting up a remote FPGA lab at our university and need to prevent multiple students from accessing the same device simultaneously to avoid conflicts.

Thanks for your help.

**Comment by si458** [comment]
these are the only options currently to do with limits, you set them under `limits` under your `domains` section in `config.json`
but i dont believe the is a limit per device sadly, only per user
![image](https://github.com/Ylianst/MeshCentral/assets/765314/c1d4f1e0-eaf2-4a02-aa3e-563cebc89b48)

**Reply by si458** [reply]
we could look at implementing another option which is `MaxDeviceSessions` which you can set globally to limit each device to only say 1 session, you would need to open an enhancement in the issues for that

**Reply by JustKeonix** [reply]
That would not allow admins to oversee student activity and assist them, but it would be 90% of the way there. Perhaps there could be a user group setting like `AllowDeviceLimitsOverride` to address that. I will create an issue then. Thanks for the help!

**Comment by silversword411** [comment]
Sounds like a good sponsored addition to me
https://github.com/sponsors/si458

**Reply by si458** [reply]
https://www.si458.co.uk/2024/01/05/donation/
